plugins/sfc/renderer/l2xconn/l2xconn_renderer.go

```diff
@@ -129,35 +130,64 @@ func (rndr *Renderer) Close() error {
 func (rndr *Renderer) renderChain(sfc *renderer.ContivSFC) (config controller.KeyValuePairs) {
 	config = make(controller.KeyValuePairs)
 	var prevSF *renderer.ServiceFunction
-
 	for _, sf := range sfc.Chain {
-		// get interface names of this and previous service function
+		// get interface names of this and previous service function (work only for local SFs else return "")
 		iface := rndr.getSFInterface(sf, true)
 		prevIface := ""
 		if prevSF != nil {
 			prevIface = rndr.getSFInterface(prevSF, false)
 		}
 
-		// TODO: this works only for local pods, inter-node chains are not supported yet (will be skipped)
 		if iface != "" && prevIface != "" {
-			// cross-connect the interfaces in both directions
-			xconn := &vpp_l2.XConnectPair{
-				ReceiveInterface:  prevIface,
-				TransmitInterface: iface,
-			}
-			key := vpp_l2.XConnectKey(prevIface)
-			config[key] = xconn
-
-			xconn = &vpp_l2.XConnectPair{
-				ReceiveInterface:  iface,
-				TransmitInterface: prevIface,
-			}
-			key = vpp_l2.XConnectKey(iface)
-			config[key] = xconn
+			if rndr.checkSFNodeLocal(sf) && rndr.checkSFNodeLocal(prevSF) {
+				// The two SFs are local
+				//for local pods do: only cross-connect the interfaces in both directions
+
+				configCrossCt := rndr.crossConnectIfaces(iface, prevIface)
+				rndr.mergeConfiguration(config, configCrossCt)
+			} else {
+				if rndr.checkSFNodeLocal(sf) || rndr.checkSFNodeLocal(prevSF) { // If  one of the SFs (prevSF or SF) is local
+					// If sf local create vxlan in the two direction the nodes
+					// and cross-connect each SF and prev SF to these vxlans else do the same with prefSF
+
+					// allocate custom vxlan vni for the actual SFC instance(reconized by SFC name, SFC instance number), if not always alocated else get it
+					// those vxlan vnis are >=5000
+					vni, err := rndr.IPAM.AllocateVNI(sfc.Name, 1) // TODO: handle chains with multiple pod instances per service function? untill now only one SFC instance is considered
+					if err != nil {
+						rndr.Log.Infof("Unable to allocate SFC VNI: %v", err)
+						break
+					}
+					if rndr.checkSFNodeLocal(sf) { // sf is local
+						// create vxlan and connect sf to vxlan in both directions
+						// direction 1
+						vxlanName, vxlanConfig := rndr.vxlanSFC(sf, prevSF, vni)
+						rndr.mergeConfiguration(config, vxlanConfig)
+
+						xconnect := rndr.connectIfaces(iface, vxlanName)
```

Comment:
why are you not using `crossConnectIfaces` here?

Reply:
Ops yeah it is possible to use the crossConnectIfaces. I have mismatched the code at the beginning (by doing a lot of change here).  do you want that I redo it with the crossConnectIfaces in a new PR?